source/target/realtek/rtl8195am/target.c

```diff
@@ -0,0 +1,37 @@
+/**
+ * @file    target.c
+ * @brief   Target information for Realtek RTL8195AM
+ *
+ * DAPLink Interface Firmware
+ * Copyright (c) 2016-2017, Realtek Semiconductor Corp., All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "target_config.h"
+
+// The file flash_blob.c must only be included in target.c
+#include "flash_blob.c"
+
+// target information
+target_cfg_t target_device = {
+    .sector_size    = KB(4),
+    .sector_cnt     = MB(2) / KB(4),
+    .flash_start    = 0x00000000,
```

Comment:
Reading the other comments, if the flash really starts at 980000, then should this not be that, instead of 0?

Reply:
Sorry for the delay. 
0x0 is the logical address but 0x98000000 is the physical address on chip. 
Also because of there are reserved flash memory on RTL8195AM, it requires to customize the function "target_flash_program_page" from "DAPLink\source\daplink\interface\target_flash.c".

1, Declare "weak" for "target_flash_program_page" at "DAPLink\source\daplink\interface\target_flash.c" 
2, Write a new "target_flash_program_page" function at "DAPLink\source\target\realtek\rtl8195am\target.c", for both physical address issue and reserved flash memory issue. 

This solution has the minimum changes in "target_flash.c" and able to fix the automation problem. Please give some suggestions about if it is able to be approved?

Reply:
Hi @M-ichae-l, recommend making use of the Verify flash algo function. When present it could be used instead of reading the contents off of the device. Info on it can be found here: 
http://www.keil.com/support/man/docs/ulink2/ulink2_alg_func.htm

Reply:
I have verified the flash algo and tried to modified. The reserved flash memory issue still appears. I suggested that it is necessary to customize the "target_flash_program_page" function.

Reply:
Hi @c1728p9 ,  the DAPlink does not use the Verify. Even I have made use of the Verify flash algo function.

Reply:
Hi, @c1728p9 . Would you like to give some suggestions about this issue?

Reply:
Support for the Verify flash algo function needs to be added to DAPLink

Reply:
Hi, @c1728p9 .
Should we wait the Verify added to DAPLink? Then process this PR. Thank you!

Reply:
Hi Russ @c1728p9  Could you please help confirm @M-ichae-l  question about adding Verify to DAPLink? Thanks.

Reply:
@c1728p9 any suggestion how Realtek should proceed with this? Thx

Reply:
@c1728p9 do you have time to help with this ?

Reply:
@jupe I don't have time currently as I am focused on other tasks. @flit may be able to help you here.

Reply:
@samchuarm 
Hi everyone, Could I know when is the Verify flash algo function will be added?

Reply:
Hi Chris @flit ,  can we resume collaborating with Realtek on getting the PR in? Thanks.